openprescribing/matrixstore/tests/test_read_existing_file.py

```diff
@@ -0,0 +1,105 @@
+import random
+import os
+import sqlite3
+import warnings
+
+from django.test import SimpleTestCase
+
+import numpy
+
+from matrixstore.matrix_ops import sparse_matrix, finalise_matrix
+from matrixstore.serializer import (
+    deserialize, serialize, serialize_compressed
+)
+
+
+class TestReadExistingFile(SimpleTestCase):
+    """
+    This tests that we can still correctly read serialized matrices (sparse,
+    dense, compressed and uncompressed) from an SQLite fixture created by a
+    previous version of the software. This should catch any backwards
+    incompatibilities introduced by upgrading our dependencies, in particular
+    PyArrow. The fixture file can be regenerated by running the test with the
+    environment variable `MATRIXSTORE_CREATE_FIXTURE` set to some non-empty
+    value.
+    """
+
+    def setUp(self):
+        self.random = random.Random()
+        self.random.seed(1204)
+
+    def test_values_can_be_read_correctly(self):
+        data = {
+            'sparse_int': self.make_matrix(sparse=True, integer=True),
+            'sparse_float': self.make_matrix(sparse=True, integer=False),
+            'dense_int': self.make_matrix(sparse=False, integer=True),
+            'dense_float': self.make_matrix(sparse=False, integer=False),
+        }
+        for key, value, expected_value in self.get_values(data):
+            self.assertIsInstance(value, type(expected_value))
+            if hasattr(value, 'todense'):
+                value = value.todense()
+                expected_value = expected_value.todense()
+            self.assertEqual(value.tolist(), expected_value.tolist())
```

Comment:
This is a good test to have.  I think it'd be easier to understand if `create_fixture()` was called separately.

Also, could the fixture contain metadata about the environment it was created in?   (eg Python version, OS version, library versions)